test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java

```diff
@@ -1048,7 +1055,114 @@ protected boolean includeEffectiveAccess(ClassList classes, ClassDescription cla
         new CreateSymbols().createSymbols(null, symbolsDesc.toAbsolutePath().toString(), classDest, 0, "8", "", modules.toString(), modulesList.toString());
     }
 
+    @Test
+    void testModuleMainClass() throws Exception {
+        ClassFile cf = ClassFile.of();
+        ToolBox tb = new ToolBox();
+        String testClasses = System.getProperty("test.classes");
+        Path output = Paths.get(testClasses, "test-data" + i++);
+        deleteRecursively(output);
+        Files.createDirectories(output);
+        Path ver9Jar = output.resolve("9.jar");
+        compileAndPack(output,
+                       ver9Jar,
+                       classesDir -> {
+                           try {
+                               Path moduleInfo = classesDir.resolve("module-info.class");
+                               byte[] newClassData =
+                                       cf.transform(cf.parse(moduleInfo),
+                                                    (builder, element) -> {
+                                                        builder.with(element);
+                                                        if (element instanceof ModuleAttribute) {
+                                                            builder.with(ModuleMainClassAttribute.of(ClassDesc.of("main.Main")));
+                                                        }
+                                                    });
+                               try (OutputStream out = Files.newOutputStream(moduleInfo)) {
+                                   out.write(newClassData);
+                               }
+                           } catch (IOException ex) {
+                               throw new UncheckedIOException(ex);
+                           }
+                       },
+                       """
+                       module m {
+                       }
+                       """,
+                       """
+                       package main;
+                       public class Main {}
+                       """);
+
+
+        Path ctSym = output.resolve("ct.sym");
+
+        deleteRecursively(ctSym);
+
+        CreateSymbols.ALLOW_NON_EXISTING_CLASSES = true;
+        CreateSymbols.EXTENSION = ".class";
+
+        deleteRecursively(ctSym);
```

Comment:
Is double deletion needed for some kind of race condition safety?

Reply:
I don't think this is needed. Thanks for noticing, removed.